src/datastructures.jl

```diff
@@ -201,11 +298,15 @@ Returns the `system` field of a `EnergySystemDesign` `design`.
 get_system(design::EnergySystemDesign) = design.system
 
 """
-    get_system_node(design::EnergySystemDesign)
+    get_element(design::EnergySystemDesign)
 
 Returns the system node (i.e. availability node for areas) of a `EnergySystemDesign` `design`.
 """
-get_system_node(design::EnergySystemDesign) = design.system[:node]
+function get_element(design::EnergySystemDesign)
+    if !isnothing(design.parent)
```

Comment:
The `!isnothing(design.parent)` should be included for performance as an additional argument,, _i.e._ 

```julia
function get_element(design::EnergySystemDesign, design.parent::Nothing)
end
function get_element(design::EnergySystemDesign, design.parent)
    return design.system[:node]
end
```
I am not fully certain, but I think it improves the speed of the program.

Reply:
do you here mean do have parent as an input argument? Or is it possible to have signatures specify the types of the arguments?